### PR TITLE
azure/blob-storage: new module

### PR DESCRIPTION
**Pull request by samkahchiin**

### CODE REVIEW

File: azure/blob-storage/variables.tf [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    change_feed_enabled           = optional(bool, true)
+    change_feed_retention_in_days = optional(number, 14)
+    last_access_time_enabled      = optional(bool, true)
+  })
+  default = {}
+}
+
+variable "public_network_access_enabled" {
+  type    = bool
+  default = true
```
Comment (swiknaba) [inline azure/blob-storage/variables.tf:90]: Do we in most cases need public access? Consider defaulting to `false` here, so people make a conscious decision when enabling it.

File: azure/blob-storage/variables.tf [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+# https://learn.microsoft.com/en-us/azure/storage/common/storage-redundancy
+variable "account_replication_type" {
+  type = string
+  # LRS: Least expensive. copies your data synchronously three times within a single physical location in the primary region
+  # ZRS: copies your data synchronously across three Azure AZs in the primary region.
+  # # Secondary Region
+  # GRS: LRS (Primary Region) + LRS in secondary region.
+  # GZRS: ZRS (Primary Region) + LRS in secondary region
+  default = "LRS"
```
Comment (swiknaba) [inline azure/blob-storage/variables.tf:70]: `GZRS` costs roughly twice as much as `LRS`, however, if we store e.g. 100 GiB, the cost difference is 4€ vs. 2€, which is minimal.

I thus suggest we go with `GZRS` as default value for higher reliability.

**Review by swiknaba — APPROVED** [review]
Overall, this looks great to me. I left two considerations regarding some default values; let me know what you think.

File: azure/blob-storage/variables.tf (hunk 1 above)
Comment (samkahchiin) [inline azure/blob-storage/variables.tf:90]:
```suggestion
  default = false
```

File: azure/blob-storage/variables.tf (hunk 2 above)
Comment (samkahchiin) [inline azure/blob-storage/variables.tf:70]:
```suggestion
  default = "GZRS"
```

**Pull request merged**